LazyCategories/gap/LazyCategory.gi

```diff
@@ -795,6 +797,57 @@ InstallMethod( LazyCategory,
     
     print := IsIdenticalObj( CAP_INTERNAL_RETURN_OPTION_OR_DEFAULT( "print", false ), true );
     
+    AddIsWellDefinedForObjects( D,
+      function( D, a )
+        local genesis, operation_name, arguments;
+        
+        # object must have a genesis or an evaluated cell
+        if not (HasGenesisOfCell( a ) or HasEvaluatedCell( a )) then
+            
+            return false;
+            
+        fi;
+        
+        if HasGenesisOfCell( a ) then
+            
+            genesis := GenesisOfCell( a );
+            
+            # genesis must be a pair
+            if not (IsList( genesis ) and Length( genesis ) = 2) then
+                
+                return false;
+                
+            fi;
+            
+            operation_name := genesis[1];
+            arguments := genesis[2];
+            
+            # operation_name must be the name of a CAP operation which returns an object
+            if not (IsString( operation_name ) and IsBound( CAP_INTERNAL_METHOD_NAME_RECORD.(operation_name) ) and CAP_INTERNAL_METHOD_NAME_RECORD.(operation_name).return_type = "object") then
+                
+                return false;
+                
+            fi;
+            
+            # arguments must be a list of arguments of the CAP operation
+            if not (IsList( arguments ) and Length( arguments ) = Length( CAP_INTERNAL_METHOD_NAME_RECORD.(operation_name).filter_list )) then
+                
+                return false;
+                
+            fi;
+            
+        fi;
+        
+        if not IsWellDefinedForObjects( C, EvaluatedCell( a ) ) then
```

Comment:
This line will enforce evaluation, is this desired?

Reply:
I'm not sure about "desired" but at least it was intentional: `IsWellDefinedForObjects` was previously installed automatically via `create_func_bool` which evaluates, so I thought I keep this backwards compatible.

Thinking more about this, we could also do recursive checks on `arguments` which would at least give some weak guarantees that the evaluated object will be well-defined. To have stronger guarantees, we would also have to evaluate (full) prefunctions which might cause evaluation again. And all of this might be a bit tedious to implement.

Some background: The reason why I created this PR are the following two assertions:
```
IsEqualForObjects( C, EvaluatedCell( Source( phi ) ), Source( EvaluatedCell( phi ) ) )
IsEqualForObjects( C, EvaluatedCell( Range( phi ) ), Range( EvaluatedCell( phi ) ) )
```
This defines what kind of morphisms are allowed. Everything else is just technical stuff which is not too important for me right now^^